docs/lesson-plans/week-1.md

```diff
@@ -0,0 +1,720 @@
+descriptions of the files right now. You'll learn a lot more about everything below in the 
+coming weeks, so don't worry if you don't know what we're saying.
+
+#### Common Files
+
+.gitignore is a file containing regular expressions to describe files we don't want Git
+to keep track of when we make commits or push to GitHub.
+
+>A regular expression (regex) is a syntax used to define text matching the given pattern. 
+>Check out [regexr](https://regexr.com/) and [regex101](https://regex101.com/) for some 
+>common regexes, descriptions, and a playground to test some out.
+
+LICENSE is a file detailing licensing information. In our case, we're using the GPL-3.0
+License, which means anyone can use or modify our code for whatever they want (with some
+conditions). You'll rarely have to deal with this in practice, but we wanted to add one
+so that you can see what it looks like.
+
+README.md is sort of like the introduction file. It's written in a format called 
+[Markdown](https://en.wikipedia.org/wiki/Markdown) (the .md filetype), which allows for 
+simple and easy formatting. Usually this file has descriptions of the software, instructions
+for setting it up, and any other information the developer(s) would like to provide to the
+end user.
+
+#### Frontend
+
+node_modules/ is a directory which houses all of the JavaScript packages you'll need. You should
+never have do anything with that directory (other than deleting it if you're getting weird issues,
+but that should be __*RARE*__), since everything having to do with this directory will be handled
+for you. This will appear after you've run `npm install` later on, which is a program to download and
+handle external dependencies and packages to be used in your code. This
+directory is also usually not included in a Git repository.
+
+public/ is a directory to store unique files and images like manifest files, HTML files, or
+special image files. They're put here for easy reference and access.
+
+src/ is the directory with your actual project files. All of your React components,
+views, tests, and code will be inside here.
+
+package.json is a file containing dependency/package information, scripts to be run using npm,
+and other data about your project.
+
+package-lock.json holds information on requested dependencies, which dependencies are required for 
+the requested dependencies, and special information that should rarely change. You should never have 
+to do anything with this file, and
+it should **_definitely_** be included in a Git repository.
+
+tsconfig.json holds information about your Typescript configuration.
+
+#### Backend
+
+pom.xml is a file with project build instructions, dependencies, and other project information.
+
+api/ is a directory handling routing information and data transfer objects (DTOs).
+
+persist/ is a directory for handling and saving data.
+
+service/ is the entry point for your project, so you'll start it up from there. It handles
```

Comment:
"entry point" can mean a lot of different things. I would just say "holds the main method for your project"

Reply:
Added.